inst/templates/ad_advs.R

```diff
@@ -0,0 +1,377 @@
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+windows <- tribble(
+  ~AVISIT,     ~AVISITN,
+  "Screening",        0,
+  "Day 1",            1,
+  "6 Months",         2,
+  "12 Months",        3
+)
+
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    )
+  ) %>%
+  derive_vars_merged(
+    dataset_add = windows,
+    by_vars = exprs(AVISIT)
+  )
+
+## Add PARAM/PARAMN ----
+advs <- advs %>%
+  # Derive PARAM and PARAMN
+  derive_vars_merged(dataset_add = select(param_lookup, -VSTESTCD), by_vars = exprs(PARAMCD))
+
+## Merge ADVS to the chosen Growth metadata as an input to meta_criteria ----
+## Calculate z-scores/percentiles
+## Calculate Weight for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated.
+advs_wgt_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = weight_for_age,
+  parameter = VSTESTCD == "WEIGHT",
+  set_values_to_sds = exprs(
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "WTASDS" ~ 5,
+      PARAMCD == "WTAPCTL" ~ 6,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("WTASDS", "WTAPCTL"))
+
+## Calculate BMI for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_bmi_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = bmi_for_age,
+  parameter = VSTESTCD == "BMI",
+  set_values_to_sds = exprs(
+    PARAMCD = "BMISDS",
+    PARAM = "BMI-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "BMIPCTL",
+    PARAM = "BMI-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "BMISDS" ~ 7,
+      PARAMCD == "BMIPCTL" ~ 8,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("BMISDS", "BMIPCTL"))
+
+## Calculate Head Circumference for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated for z-score and percentile in final dataset.
+advs_hdc_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = who_hc_for_age,
+  parameter = VSTESTCD == "HDCIRC",
+  set_values_to_sds = exprs(
+    PARAMCD = "HDCSDS",
+    PARAM = "HDC-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "HDCPCTL",
+    PARAM = "HDC-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "HDCSDS" ~ 9,
+      PARAMCD == "HDCPCTL" ~ 10,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("HDCSDS", "HDCPCTL"))
+
+## Calculate for Height/Length ----
```

Comment:
also for this step Samia left a note in the outline doc: For-Height/Length SDS/Percentiles
Create a temporary Height variable in cm (e.g. HGTTMP) and add the height/length at each time point